datafusion/functions-window/src/rank.rs

```diff
@@ -39,37 +39,76 @@ use field::WindowUDFFieldArgs;
 define_udwf_and_expr!(
     Rank,
     rank,
-    "Returns rank of the current row with gaps. Same as `row_number` of its first peer"
+    "Returns rank of the current row with gaps. Same as `row_number` of its first peer",
+    Rank::basic
 );
 
+// define_udwf_and_expr!(
```

Comment:
@jcsherin while defining the other udwf like this, it is not working, giving an error during compile time.
Am I going in the right direction ?

Thanks !

Reply:
The first argument needs to be a distinct identifier. I guess something like this will work,

```rust
define_udwf_and_expr!(BasicRank, rank, "...", Rank::basic);
define_udwf_and_expr!(PercentRank, rank, "...", Rank::percent_rank);
define_udwf_and_expr!(DenseRank, rank, "...", Rank::dense_rank);
```